src/utils/index.js

```diff
@@ -1,68 +1,123 @@
+
+const getOperatorsArg = key => {
+  const match = key.match(operatorsRegexp);
+
+  if (!match) return {};
+
+  const [, name, argsStr] = match;
+
+  const args = argsStr.split(',')
+    .map(item => item.trim().replace(/^'|'$/g, ''))
+    .filter(Boolean);
+
+  return {name, args};
 };
 
-module.exports.getPath = (obj, path) => {
-  let varsPath = path.split(".");
-  let currval = obj;
-  if (
-    varsPath.some(varsPathItem => {
-      const arrMatch = varsPathItem.match(/^([^\[|^\]]*)\[([^\]]+)\]$/);
-      if (!arrMatch) currval = currval[varsPathItem];
-      else {
-        const idx = parseInt(arrMatch[2]);
-        if (!isNaN(idx)) currval = currval[arrMatch[1]][idx];
-        else {
-          currval = currval[arrMatch[1]].map(el => {
-            return getPath(arrMatch[2], el);
-          });
-        }
+const getBracketsArg = path => {
+  const brackets = {
+    '(': 0,
+    '[': 0
+  };
+
+  let arg = '';
+  brackets[path[0]]++;
+  [...path.substring(1)].some(char => {
+    if (char in brackets) brackets[char]++;
+    if (char === ')') brackets['(']--;
+    if (char === ']') brackets['[']--;
+    if (brackets['('] + brackets['['] === 0) return true;
+    arg += char;
+    return false;
+  });
+
+  return arg;
+};
+
+module.exports.getPath = (root, path) => {
+  path = (path || '').trim();
+
+  // remove leading dot (.) in case dot after closing bracket e.g. ['root'].value
+  path = path.startsWith('.') ? path.substring(1).trim() : path;
+
+  if (!path.trim()) return root;
+
+  const match = path.match(openersRegexp);
+
+  if (!match) return root[path];
+
+  const {0: delimiter, index: pos} = match;
+
+  let nextPath;
+  let key;
+
+  switch (delimiter) {
+    case '.':
+      key = path.substring(0, pos);
+      nextPath = path.substring(key.length + 1);
+      return this.getPath(root[key], nextPath);
+    case '[':
+      if (pos > 0) {
+        key = path.substring(0, pos);
+        nextPath = path.substring(key.length);
+        return this.getPath(root[key], nextPath);
       }
-      return !currval;
-    })
-  )
-    return null;
-  return currval;
+
+      key = getBracketsArg(path);
+      nextPath = path.substring(key.length + 2);
+
+      key = key.trim().replace(/^'|'$/g, '');
+
+      const {name, args} = getOperatorsArg(key);
+      if (!name) return this.getPath(root[key], nextPath);
+
+      const nextRoot = operators[name](root, ...args);
+      return this.getPath(nextRoot, nextPath);
+  }
+};
+
+const toJson = (mapping) => {
+  if (mapping.startsWith('{') && mapping.endsWith('}')) {
+    const [key, value] = mapping
+      .substring(1, mapping.length - 1)
+      .split(/:(.*)/)
+      .map(str => str.trim())
+      .filter(Boolean);
+
+    return {[key]: toJson(value)};
+  }
+
+  return mapping;
 };
 
 module.exports.transform = (root, mapping) => {
-  const transformed = {};
+  let transformed = {};
+
+  if (typeof mapping === 'string') {
```

Comment:
If mapping is a string - we don't need to try and make it an object

Reply:
Why not? if I'm coming from a path like this: `root[map({id: 'root.id'})]` I will get the mapping ```"{id: 'root.id'}"``` (as a string)